### PR TITLE
MRG: refactoring for pydicom 1.0

### PR DESCRIPTION
**Pull request by matthew-brett**
Make tests pass against pydicom 1.0.

Refactor CI to test pydicom master, as suggested by pydicom folks.

Update minimum pydicom dependency to 0.9.9 (previous versions no longer install).

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -34,29 +33,29 @@ matrix:
     # Absolute minimum dependencies
     - python: 2.7
       env:
-        - DEPENDS="numpy==1.7.1" PYDICOM=0
+        - DEPENDS="numpy==1.7.1"
```
Comment (effigies) [inline .travis.yml]: Wasn't the idea here to make sure that we didn't break when pydicom was absent? Do we want to move to a pydicom dependency?

Comment (effigies) [reply .travis.yml]: Wow, brain failure. I see this replaces the pydicom dependency above. Carry on.

**Comment by effigies** [comment]
This looks reasonable. The AppVeyor build seems to have failed for its own reasons (I can't restart it; not sure if that's a thing in AppVeyor).

File: .travis.yml (hunk 1 above)
Comment (matthew-brett) [reply .travis.yml]: With this matrix entry, Pydicom isn't in the DEPENDS variable, and doesn't get installed.  On the way, the tests did in fact fail when Pydicom wasn't present, I had to fix the code to make it work again.  Have a look at the Travis-CI log to confirm this entry does not install Pydicom...

**Comment by matthew-brett** [comment]
I reverted a couple of the modifications, because pydicom has fixed them already...

**Comment by matthew-brett** [comment]
Thanks Chris for the review.

**Pull request merged**